### PR TITLE
Minor: Add tests to confirm Meter creation behavior

### PR DESCRIPTION
**Pull request by cijothomas**
Very basic test. Need to eventually get to more solid ones, but this is a start to ensure things continue to work while I am working through refactoring some Metric sdk internals.

### CODE REVIEW

File: opentelemetry-sdk/src/metrics/meter_provider.rs [hunk 1]
```diff
@@ -433,4 +434,28 @@ mod tests {
         // Now the shutdown should be invoked
         assert!(reader.is_shutdown());
     }
+
+    #[test]
+    fn same_meter_reused_same_scope() {
+        let provider = super::SdkMeterProvider::builder().build();
+        let _meter1 = provider.meter("test");
```
Comment (TommyCpp) [inline opentelemetry-sdk/src/metrics/meter_provider.rs:441]:
```suggestion
        // provider should reuse the meter if the name is the same
        let _meter1 = provider.meter("test");
```

File: opentelemetry-sdk/src/metrics/meter_provider.rs [hunk 2]
```diff
@@ -433,4 +434,28 @@ mod tests {
...
+        let _meter2 = provider.meter("test");
+        assert_eq!(provider.inner.meters.lock().unwrap().len(), 1);
+        let _meter3 =
```
Comment (TommyCpp) [inline opentelemetry-sdk/src/metrics/meter_provider.rs:444]:
```suggestion
        // don't reuse the meter if the version and schema URL is different
        let _meter3 = 
```

File: opentelemetry-sdk/src/metrics/meter_provider.rs [hunk 3]
```diff
@@ -433,4 +434,28 @@ mod tests {
...
+            provider.versioned_meter("test", Some("1.0.0"), Some("http://example.com"), None);
+        let _meter4 =
+            provider.versioned_meter("test", Some("1.0.0"), Some("http://example.com"), None);
+        let _meter5 =
+            provider.versioned_meter("test", Some("1.0.0"), Some("http://example.com"), None);
```
Comment (TommyCpp) [inline opentelemetry-sdk/src/metrics/meter_provider.rs:449]: Maybe also add one for schema URL? Not a blocker

**Review by lalitb — APPROVED** [review]

File: opentelemetry-sdk/src/metrics/meter_provider.rs (hunk 3 above)
Comment (cijothomas) [reply opentelemetry-sdk/src/metrics/meter_provider.rs:449]: Yes, a ton of new tests will be added soon! 
There are more considerations for Meter identities.. not quite sure how to tackle them all. Adding basic tests now, just to be alerted myself while refactoring!

**Pull request merged**